src/import_whatsapp.js

```diff
@@ -207,16 +204,61 @@ const processText = (text) => {
 		if (!Object.keys(senders).includes(message.sender)) {
 			senders[message.sender] = getSenderColour(senders);
 		}
-		let location = locationRegex.exec(message.content);
+		var location = locationRegex.exec(message.content);
 		if (location) {
 			message.location = {
 				lat: parseFloat(location[1]),
 				long: parseFloat(location[2]),
 			};
 		}
+		let imgFileMatches = [...message.content.matchAll(imgFileRegex)];
+		if (imgFileMatches.length > 0) {
+			let imgMatches = [];
+			for (const match of imgFileMatches) {
+				imgMatches.push(match[1]);
+			}
+			message.imgFilenames = imgMatches;
+		}
+		message.content = cleanMsgContent(message.content, location, imgFileRegex);
 		messages.push(message);
 	});
+	return [messages, senders];
+};
+
+// Regex to match google maps location and capture lat (group 1) and long (group 2)
+const locationRegex =
+	/: https:\/\/maps\.google\.com\/\?q=(-?\d+\.\d+),(-?\d+\.\d+)/; //Without 'location' to be universal - the word in the export file changes based on WA language
+
+const setImgMsgRegex = (fileType) => {
+	let messageRegex;
+	let imgFileRegex;
+	// Regex matches a single message including newline characters,
+	// stopping when new line starts with date or text ends
+	// also accounts for if the datetime is wrapped in brackets and has s
+	// Capture group 1 = date, group 2 = time, group 3 = sender, group 4 = message content
+	// this has been tweaked for each format but gives the same output
+	// TODO: pass along which format to the map so it knows how to find images
+	if (fileType.match(/\[\d{2}/)) {
+		console.info("ios format");
+		// iOS format
+		messageRegex =
+			/\[(\d{2}\/\d{2}\/\d{4}),\s(\d{1,2}:\d{2}:\d{2}\s(?:AM|PM))\]\s(.*?):\s(.+?)(?=\n\[|$)/gs;
+		imgFileRegex = /<attached: (\d+-[\w\-_]+\.(jpg|jpeg|png|gif))>/gim;
+	} else if (fileType.match(/\d{2}\//)) {
+		console.info("android format");
+		// Android format
+		messageRegex =
+			/(\d{2}\/\d{2}\/\d{4}),?\s(\d{1,2}:\d{2})(?:\s?(?:AM|PM|am|pm))?\s-\s(.*?):[\t\f\cK ]((.|\n)*?)(?=(\n\d{2}\/\d{2}\/\d{4})|$)/g;
+		// Regex to match and capture image filenames in messages
+		imgFileRegex = /\b([\w\-_]*\.(jpg|jpeg|png|gif))\s\(file attached\)/gim;
+	} else {
+		console.error("Unknown file format");
+		return [null, groupName];
```

Comment:
Why return groupName here?

Reply:
this was leftover from a previous thing - good spot!